runtime/Go/antlr/dfa.go

```diff
@@ -5,7 +5,7 @@ import "sort"
 type DFA struct {
 	atnStartState DecisionState
 	decision      int
-	_states       map[string]*DFAState
+	states        map[string]*DFAState
```

Comment:
>IMHO point 2 is important for users who import the runtime API and maybe lookup code in order to quickly understand behaviour (the last one is one of Go's strong suits, IMHO). Also I think 2 is important for 3, although 3 should not have any importance right now.

I think your second point (quick understanding) is the main one: the virtue of idiomatic code is readability and intelligibility, which ease maintenance. We should write the code assuming someone else will maintain it. The runtime code is (IIUC) only used under the hood by generated code, so it's really just for our (the implementors) benefit.

---

My assumption so far has been that the templates form a contract between ANTLR and the language targets, and implementing those templates are all that's required, now and in the future. (Please correct me if I'm wrong!) If a new required template is added, or an existing template's contract is changed in a backward-incompatible way, all the language targets would break, so it wouldn't be just Go's problem. So we just need to fill in the template bodies with whatever Go code gets the job done.

To preserve backward compatibility, future ANTLR innovations would have to either be opt-in (like the XSL-like querying, IIUC) or added to all language targets at once. I'm not sure what we could do to anticipate and accommodate either case, other than just write good code now. The current Go template implementations seem comparable to those of other language targets.

None of the templates seem complicated enough to warrant exact/close source code matching, but perhaps I'm missing some complication there. The templates seem to be granular enough that there isn't much flexibility possible in their implementation to better accommodate future innovations. It seems like innovations would come in the form of new templates, rather than changing existing ones, or backward-incompatible changes.

Did I understand/address your concerns about 1 and 4? I agree it's worth it to plan ahead for these considerations.